generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/Combined.java

```diff
@@ -0,0 +1,84 @@
+package com.scottlogic.deg.generator.walker.reductive;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+public class Combined<T> {
```

Comment:
Minor-ish comment , could this extend `Optional`?

Reply:
Could this be renamed to better clarify what information/purpose it portrays, `Combined` is a bit ambiguous, how about `Merged`?